src/sonic-eventd/rsyslog_plugin/main.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <unistd.h>
+#include "rsyslog_plugin.h"
+#include "syslog_parser.h"
+
+using namespace std;
+
+void showUsage() {
+    cerr << "Usage for rsyslog_plugin: " << " <option(s)> SOURCES\n"
+        << "Options:\n"
+        << "\t-r,required,type=string\t\tPath to regex file"
+        << "\t-m,required,type=string\t\tYANG module name of source generating syslog message"
+        << endl;
+}
+
+int main(int argc, char** argv) {
+    string regex_path;
+    string module_name;
+    int option_val;
+
+    while((option_val = getopt(argc, argv, "r:m:")) != -1) {
+        switch(option_val) {
+            case 'r':
+                if(optarg != NULL) {
```

Comment:
No need for this check. getopt ensures non null optarg. You may test it.

Reply:
Done